File: problems/819-most-common-word/solution.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        Set<String> bannedSet = new HashSet<>(Arrays.asList(banned));
        Map<String, Integer> wordCounts = new HashMap<>();
        String[] words = paragraph.toLowerCase().split("[^a-zA-Z]+");
        for (String word : words) {
            if (!bannedSet.contains(word)) {
                wordCounts.put(word, wordCounts.getOrDefault(word, 0) + 1);
            }
        }
        String mostCommonWord = "";
        int maxCount = 0;
        for (Map.Entry<String, Integer> entry : wordCounts.entrySet()) {
            if (entry.getValue() > maxCount) {
                maxCount = entry.getValue();
                mostCommonWord = entry.getKey();
            }
        }
        return mostCommonWord;
    }
}

File: problems/819-most-common-word/solution.py
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        import re
        from collections import Counter
        words = re.findall(r'\b\w+\b', paragraph.lower())
        banned_set = set(banned)
        word_counts = Counter(word for word in words if word not in banned_set)
        return word_counts.most_common(1)[0][0]